src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs

```diff
@@ -39,14 +40,20 @@ private void RoundTrip(Action<ObjectWriter> writeAction, Action<ObjectReader> re
         {
             using var stream = new MemoryStream();
 
-            using (var writer = new ObjectWriter(stream, leaveOpen: true))
+            ImmutableArray<object> keepAliveObjects;
+            using (var writer = new ObjectWriter(stream, leaveOpen: true, canKeepObjectsAlive: true))
             {
                 writeAction(writer);
+
+                keepAliveObjects = writer.TakeKeepAliveObjects();
             }
 
             stream.Position = 0;
             using var reader = ObjectReader.TryGetReader(stream);
             readAction(reader);
+
+            foreach (var obj in keepAliveObjects)
+                GC.KeepAlive(obj);
```

Comment:
wouldn't: `GC.KeepAlive(keepAliveObjects)` suffice here?  feels weird to do this loop.

Reply:
Technically yes. But technically `keepAliveObjects` is a value type so it's an analyzer red flag.

Reply:
i think i prefer the 'technically' :)  TBH, my understanding of GC.KeepAlive is that's is effectively a no-op, but one that can't be eliminated.  i.e. it's just used as  marker so that the runtime doesn't go 'keepAliveObjects' isn't going to be used, i can GC it up front even while it's on the stack.  It's effectively just a lifetime indicator that is guaranteed to not be "smartly optimized away".  

So in this case, all we want to say is that whatever keepAliveObjects points to never goes away.  I think `GC.KeepAlive(keepAliveObjects)` expresses that much more clearly and idiomatically.  (This is esp. the case as i'm fairly certain we don't have this looping form anywhere else in teh codebase).

Reply:
The value type was eliminated so the loop was removed too